src/grid.jl

```diff
@@ -0,0 +1,90 @@
+# Grid types
+export Node, Cell, Grid
+
+# Cell typealias
+export Line, QuadraticLine,
+       Triangle, QuadraticTriangle, Quadrilateral, QuadraticQuadrilateral,
+       Tetrahedron, QuadraticTetrahedron, Hexahedron, QuadraticHexahedron
+
+# Grid utilities
+export cells, nodes, cell_coordinates, n_cells, n_nodes
+
+#########################
+# Main types for meshes #
+#########################
+"""
+A `Node` is a point in space.
+"""
+immutable Node{dim, T}
```

Comment:
Maybe have an id for the `Node` type aswell? Could maybe help with the implementation of `DoFHandler`.